proposals/lambda-method-group-defaults.md

```diff
@@ -134,42 +169,82 @@ DoAction(writeInt, 3); // Error, cannot convert from anonymous delegate type to
+ explicit_anonymous_function_parameter_list
++    : explicit_anonymous_function_parameters (',' parameter_array)?
++    | parameter_array
++    ;
++
++explicit_anonymous_function_parameters
+     : explicit_anonymous_function_parameter
+       (',' explicit_anonymous_function_parameter)*
+     ;
+
+ explicit_anonymous_function_parameter
 -    : anonymous_function_parameter_modifier? type identifier
 +    : anonymous_function_parameter_modifier? type identifier default_argument?
-    ;
+     ;
 ```
+
+Same rules as for method parameters ([§14.6.2](https://github.com/dotnet/csharpstandard/blob/draft-v7/standard/classes.md#1462-method-parameters)) apply for lambda parameters:
+- A parameter with a `ref`, `out` or `this` modifier cannot have a *default_argument*.
+- A *parameter_array* may occur after an optional parameter, but cannot have a default value – the omission of arguments for a *parameter_array* would instead result in the creation of an empty array.
+
 No changes to the grammar are necessary for method groups since this proposal would only change their semantics.
 
-### Binder Changes
+### Updates of prior proposals
+
+The following addition (in bold) is required to the [function types](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md#natural-function-type) specification in a prior proposal:
+
+> The natural type of an anonymous function expression or method group is a *function_type*. A *function_type* represents a method signature: the parameter types, **default values, ref kinds, `params` modifiers**, and return type and ref kind. Anonymous function expressions or method groups with the same signature have the same *function_type*.
+
+The following addition (in bold) is required to the [delegate types](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-10.0/lambda-improvements.md#delegate-types) specification in a prior proposal:
+
+> The delegate type for the anonymous function or method group with parameter types `P1, ..., Pn` and return type `R` is:
+>
+> - if any parameter or return value is not by value, **or any parameter is optional or `params`**, or there are more than 16 parameters, or any of the parameter types or return are not valid type arguments (say, `(int* p) => { }`), then the delegate is a synthesized `internal` anonymous delegate type with signature that matches the anonymous function or method group, and with parameter names `arg1, ..., argn` or `arg` if a single parameter;
+> [...]
 
-#### Synthesizing New Delegate Types
-As with the behavior for delegates with `ref` or `out` parameters, a new natural type is generated for each lambda or method group defined with any default parameter values.
+### Binder changes
+
+#### Synthesizing new delegate types
+As with the behavior for delegates with `ref` or `out` parameters, a new natural type is generated for each lambda or method group defined with any optional or `params` parameters.
```

Comment:
>a new natural type is generated for each lambda or method group [](http://example.com/codeflow?start=67&length=63)

This text wasn't added in this PR but it's not clear what "natural type" means. Assuming this refers to the synthesized delegate type, perhaps it should say "delegate types are synthesized for lambdas or method groups with optional or `params` parameters."

Reply:
It's not clear to me either. Changed: 69d5c7ecbbbc173f9280083af57491fcfb3b7245 Thanks.